### PR TITLE
Add images to fields with rich text formatting

### PR DESCRIPTION
**Pull request by ZmagoD**
https://biosistemika.atlassian.net/browse/SCI-1203
relates: http://gitlab.biosistemika.com/scinote/billing/merge_requests/66
still needs to be tested with scinote-ee and with scinote-cloud addons!

### CODE REVIEW

File: app/assets/javascripts/sitewide/tiny_mce_file_upload_plugin.js.erb [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                                        ), 10),
+          body: [
+            {type: 'iframe', url: 'javascript:void(0)'},
+            {type: 'textbox',
+             name: 'file',
+             label: "<%= I18n.t 'tiny_mce.upload_window_label' %>",
+             subtype: 'file'},
+            {type: 'container',
+             classes: 'error',
+             html: "<p style='color: #b94a48;'>&nbsp;</p>"},
```
Comment (ZmagoD) [inline app/assets/javascripts/sitewide/tiny_mce_file_upload_plugin.js.erb]: NO tinyMCE adds a new <p> tag automatically. Now I have removed this function.

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -103,6 +103,7 @@ def generate_annotation_notification(target_user, title, message)
   end
 
   def smart_annotation_parser(text, team = nil)
+    team = nil unless team.is_a? Team
```
Comment (ZmagoD) [inline app/helpers/application_helper.rb]: we need!

File: lib/tasks/tiny_mce_asset.rake [hunk 3]
```diff
@@ -0,0 +1,10 @@
+namespace :tiny_mce_asset do
+  desc 'Remove obsolete images that were created on new steps or '\
+       'results and the step/result didn\'t get saved.'
+  task remote_obsolete_images: :environment do
```
Comment (ZmagoD) [inline lib/tasks/tiny_mce_asset.rake]: if someone generate a step and don't save it and in the process he/she creates a few images the images will get stored in our s3 but they will be obsolete see here: http://www.dictionary.com/browse/obsolete
with this task we will destroy it. I add the created_at time to be 7 days ago hopefully nobody need 7 days to edit a step description.

Comment (mz3944) [inline lib/tasks/tiny_mce_asset.rake]: :<. I know, I just said the naming of the function isn't OK. Just rename it from `remote_obsolete_images` to `remove_obsolete_images`. Do you have dyslexia?: https://www.understood.org/en/learning-attention-issues/child-learning-disabilities/dyslexia/understanding-dyslexia.

File: app/controllers/result_texts_controller.rb [hunk 4]
```diff
@@ -40,7 +43,9 @@ def create
     @result.last_modified_by = current_user
 
     respond_to do |format|
-      if (@result.save and @result_text.save) then
+      if (@result.save and @result_text.save)
```
Comment (mz3944) [inline app/controllers/result_texts_controller.rb]: We said we won't be using `and` no more... And you can also remove parenthesis.

**Comment by mz3944** [comment]
I have just 2 more comments, check them out.

**Comment by mz3944** [comment]
Merge away! Hurrey!

**Pull request merged**